src/main/java/io/vavr/collection/Iterator.java

```diff
@@ -1722,10 +1722,16 @@ public U next() {
         if (!hasNext()) {
             return Tuple.of(empty(), empty());
         } else {
-            final Stream<T> that = Stream.ofAll(this);
```

Comment:
There is also `partition` method implemented by `Stream` which works wrongly either.

I do not know another places. Maybe @danieldietrich could point them out.

Reply:
I was not aware of the problem of `Stream#partition`. Thanks for pointing it out! :bow: I checked all data-structures in VAVR, apart from Stream, all other implementations work correctly. About the Stream itself, I'm afraid that I don't know how to fix it and I'm not sure if it's really a bug. In Scala 2.13.2, the Stream implementation uses the predicates twice ([link](https://github.com/scala/scala/blob/v2.13.2/src/library/scala/collection/immutable/Stream.scala#L151)) as below:

```scala
override def partition(p: A => Boolean): (Stream[A], Stream[A]) = (filter(p(_)), filterNot(p(_)))
```

If I modify your example in issue #2559 by changing Set into Stream, you will see result in Scala as follow:

```scala
  "Stream" should "partition correctly" in {
    val fruitsToEat = Stream("apple", "banana")
    val partition = fruitsToEat.partition(name => biteAndCheck(name))

    partition._1 shouldEqual Stream()
    partition._2 shouldEqual Stream()  // not Stream("apple", "banana")

    fruitsBeingEaten.get("apple").get.name shouldEqual "apple"
    fruitsBeingEaten.get("apple").get.bites shouldEqual 2  // not 1
    fruitsBeingEaten.get("banana").get.name shouldEqual "banana"
    fruitsBeingEaten.get("banana").get.bites shouldEqual 2  // not 1
  }
```

So I would say the VAVR is aligned with Scala on Stream's behaviors. Also, my current approach (ArrayList) does not fit the Stream requirement, because Stream is lazy sequence of elements which may be infinitely long. So I don't know how to use only one predicate to achieve that... So my suggestion is let's keep stream as it is and avoid modifying it.

Reply:
> Stream is lazy sequence of elements which may be infinitely long

Oh sorry guys I completely forgot about it :(